### PR TITLE
[DOC] Update supported dialects for JDBC engine

### PR DESCRIPTION
**Pull request by ymZhao1001**
### _Why are the changes needed?_

As https://github.com/apache/kyuubi/pull/4000 adds the Apache Phoenix support for the JDBC engine without upgrading the configuration's description.

This PR aims to update the outdated configuration description to reflect the capabilities of the Kyuubi JDBC engine.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No.

### CODE REVIEW

**Comment by bowenliang123** [comment]
Please follow the instructions as shown in failures from CI tests.
1. run `dev/reformat` to apply styling to source code.
2. run `dev/gen/gen_all_config_docs.sh` to regenerate `settings.md`

**Comment by ymZhao1001** [comment]
> Please follow the instructions as shown in failures from CI tests.
> 
> 1. run `dev/reformat` to apply styling to source code.
> 2. run `dev/gen/gen_all_config_docs.sh` to regenerate `settings.md`

done

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -2005,7 +2005,7 @@ object KyuubiConf {
       " <li>HIVE_SQL: specify this engine type will launch a Hive engine which can provide" +
       " all the capacity of the Hive Server2.</li>" +
       " <li>JDBC: specify this engine type will launch a JDBC engine which can provide" +
-      " a MySQL protocol connector, for now we only support Doris dialect.</li>" +
+      " a MySQL protocol connector, for now we support Doris dialect and Phoenix dialect.</li>" +
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:2008]: JDBC: specify this engine type will launch a JDBC engine which can forward queries to the database system through the certain JDBC driver, for now, it supports Doris and Phoenix.

Comment (ymZhao1001) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: thanks for the review，updated。

**Comment by pan3793** [comment]
Thanks, merged to master/1.8

**Pull request closed without merging**